### PR TITLE
feat(stream,models): incremental reasoning streaming + live display pricing

### PR DESCRIPTION
**Pull request by skyscribe-yf**
## Summary

This PR adds two features and fixes a streaming bug, all built on top of the new dynamic model fetching from v0.1.1.

### Changes

**fix(stream): Emit reasoning-delta events incrementally** (`87f7f0b`)
- Previously the full thinking block was buffered until `reasoning-end`. Now each `reasoning-delta` pushes `thinking_start` / `thinking_delta` immediately so the UI renders reasoning in real time.
- Ends open text blocks on `reasoning-start` and `reasoning-delta` so thinking and text never overlap.

**feat(models): Add live display pricing for known Command Code models** (`b435005`)
- The Provider API does not expose pricing yet. Overlays a static `MODEL_COSTS` map on top of the dynamically fetched model list so pi can display per-model costs.
- Models not in the table fall back to zero cost. When the Provider API adds a `cost` field, the static table can be removed.
- Includes promotional pricing for DeepSeek V4 Pro (4× deal) and Qwen 3.7 Max (2× deal), documented in comments.

**test(pricing): Verify MODEL_COSTS entries and promotional notes** (`46c144a`)
- New `tests/test-pricing.ts` with 3 test cases:
  - Known models have non-zero pricing
  - Promotional deals are documented in source comments
  - Claude models include cache pricing (cacheRead / cacheWrite)
- Fixes `test-pi-local` to match against combined `stdout + stderr` since pi may output model listing to either stream.

**docs(changelog): Add 0.2.0 entry and pricing section to README** (`ae6e74b`)
- `CHANGELOG.md`: Added `## [0.2.0] - 2026-05-27` with 4 bullet items.
- `README.md`: Added "Pricing" section explaining the `MODEL_COSTS` overlay and how to update prices.

## Testing

- **50 tests pass** (18 pure + 3 models + 3 pricing + 12 oauth + 2 abort + 12 stream + local e2e)
- `npm run typecheck` ✅
- `npm run format:check` ✅
- `git diff --check` ✅ (no whitespace errors)

## Compliance

All changes follow `CONTRIBUTING.md` requirements:
- [x] Conventional Commits with scopes
- [x] Commit bodies explain motivation
- [x] CHANGELOG.md updated
- [x] README.md updated for user-facing changes
- [x] New features have test coverage
- [x] No `as any` / `@ts-ignore`
- [x] Formatting and linting clean

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -262,12 +251,32 @@ export function createStreamCommandCode(deps: CoreDependencies) {
 
           case "reasoning-delta": {
             endTextBlock()
-            thinkingBlock.push(stringValue(event.text) ?? "")
+            const delta = stringValue(event.text) ?? ""
+            if (thinkingIdx < 0) {
```
Comment (patlux) [inline src/core.ts:262]: This keeps a thinking block open until `reasoning-end` or the final flush. If upstream omits `reasoning-end` and starts emitting text/tool output, pi will see `text_start`/`toolcall_start` before `thinking_end` (I reproduced `thinking_start -> thinking_delta -> text_start -> ... -> thinking_end`). Please close an open thinking block before starting text or tool output, and add a regression with a delayed stream so incremental reasoning is observable before `reasoning-end`/`finish`.

File: index.ts [hunk 2]
```diff
@@ -22,6 +22,49 @@ import { getApiKey, login, refreshToken } from "./src/oauth.ts"
...
+  "claude-opus-4-6": { input: 5, output: 25, cacheRead: 0.5, cacheWrite: 6.25 },
+  "claude-sonnet-4-6": { input: 3, output: 15, cacheRead: 0.3, cacheWrite: 3.75 },
+  "claude-haiku-4-5-20251001": { input: 1, output: 5, cacheRead: 0.1, cacheWrite: 1.25 },
+  "gpt-5.5": { input: 5, output: 30, cacheRead: 0.5, cacheWrite: 0 },
+  "gpt-5.4": { input: 2.5, output: 15, cacheRead: 0.25, cacheWrite: 0 },
+  "gpt-5.3-codex": { input: 2, output: 8, cacheRead: 0.5, cacheWrite: 0 },
+  "gpt-5.4-mini": { input: 0.75, output: 4.5, cacheRead: 0.075, cacheWrite: 0 },
+  "google/gemini-3.5-flash": { input: 1.5, output: 9, cacheRead: 0.15, cacheWrite: 0 },
+  "google/gemini-3.1-flash-lite": { input: 0.25, output: 1.5, cacheRead: 0.03, cacheWrite: 0 },
+  // 4× usage deal: 75% off through May 31, 2026
```
Comment (patlux) [inline index.ts]: The current Command Code docs say the DeepSeek V4 Pro discount is permanent/no-expiry, not through May 31, 2026. Please update this note so the displayed pricing metadata doesn't go stale immediately.

File: index.ts [hunk 3]
```diff
@@ -22,6 +22,49 @@ import { getApiKey, login, refreshToken } from "./src/oauth.ts"
...
+  "deepseek/deepseek-v4-pro": { input: 0.435, output: 0.87, cacheRead: 0.003625, cacheWrite: 0 },
+  "deepseek/deepseek-v4-flash": { input: 0.14, output: 0.28, cacheRead: 0.01, cacheWrite: 0 },
```
Comment (patlux) [inline index.ts]: The DeepSeek V4 Flash cache-read rate looks off: docs list about `$0.028`/`$0.03` per 1M cache-read tokens, but this table uses `0.01`. Also, the live Provider API currently returns `xiaomi/mimo-v2.5-pro` and `xiaomi/mimo-v2.5`; since they are missing here, they display as zero-cost models. Please sync the table with the live model list/pricing docs.

File: README.md [hunk 4]
```diff
@@ -115,6 +115,15 @@ https://api.commandcode.ai/provider/v1/models
...
 For tests or local mocks, override it with `COMMANDCODE_MODELS_URL`.
 
+## Pricing
+
+Command Code does not yet expose model pricing through its Provider API. The provider ships a static cost table (`MODEL_COSTS` in `index.ts`) for known models so that pi can display per-model pricing.
+
+- Models present in `MODEL_COSTS` show their real per-million-token rates (including promotional deals like the DeepSeek V4 Pro 4× discount and Qwen 3.7 Max 2× discount).
+- Models **not** in the table fall back to zero cost. When the Provider API adds a `cost` field, the static table can be removed.
+
+To add or update a price, edit the `MODEL_COSTS` record in `index.ts` and update the corresponding test in `tests/test-models.ts`.
```
Comment (patlux) [inline README.md]: This should point to `tests/test-pricing.ts`, not `tests/test-models.ts`, since the new pricing assertions live in the new pricing test file.

**Review by patlux — CHANGES_REQUESTED** [review]
Thanks for the PR. I ran `npm test`, `npm run format:check`, and `git diff --check`; all pass locally. Requesting changes for the streaming edge case and pricing-table accuracy/completeness noted inline. ~reviewed by chatgpt-5.5 xhigh

**Comment by patlux** [comment]
Thanks @skyscribe-yf — all review feedback addressed and verified live against the Command Code API. Both features working great:

- Incremental reasoning streaming renders thinking tokens in real-time ✓
- MODEL_COSTS pricing shows correctly in the cost tracker ($0.001 for a simple query) ✓

Merging! 🎉

**Comment by patlux** [comment]
Released in [`pi-commandcode-provider@0.3.0`](https://www.npmjs.com/package/pi-commandcode-provider/v/0.3.0) — [GitHub Release](https://github.com/patlux/pi-commandcode-provider/releases/tag/v0.3.0). Thanks for the contribution! 🎉
